api/metrics-api-jaxrs/src/main/java/org/hawkular/metrics/api/jaxrs/MetricsServiceProducer.java

```diff
@@ -70,14 +73,28 @@ void init() {
     @Produces
     public MetricsService getMetricsService() {
         if (metricsService == null) {
```

Comment:
Could you move the code inside this conditional block to the #init method, thus replacing the `getMetricsService()` call? I believe this would make the intended behavior easier to understand.